core/arch/arm/kernel/delay.c

```diff
@@ -29,12 +30,14 @@
 #include <arm.h>
 #include <kernel/delay.h>
 
+#define US2CNT(m)	(((uint64_t)m * (uint64_t)read_cntfrq()) / 1000000ULL)
+
 void udelay(uint32_t us)
 {
```

Comment:
Since you're making changes to this function, why not simplify it by using the `timeout_init_us()` and `timeout_elapsed()` functions?

Reply:
ok